sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -150,7 +179,359 @@ class KmacTest : public testing::Test, public mock_mmio::MmioTest {
       setExpectedMessageByte(message, remaining);
     }
   }
+
+  void setExpectedConfig(void) {
+    EXPECT_WRITE32_SHADOWED(
+        KMAC_CFG_SHADOWED_REG_OFFSET,
+        {{KMAC_CFG_SHADOWED_KMAC_EN_BIT, config_.en},
+         {KMAC_CFG_SHADOWED_KSTRENGTH_OFFSET, config_.key_strength},
+         {KMAC_CFG_SHADOWED_MODE_OFFSET, config_.mode},
+         {KMAC_CFG_SHADOWED_MSG_ENDIANNESS_BIT, config_.msg_endianess},
+         {KMAC_CFG_SHADOWED_STATE_ENDIANNESS_BIT, config_.state_endianess},
+         {KMAC_CFG_SHADOWED_SIDELOAD_BIT, config_.sideload},
+         {KMAC_CFG_SHADOWED_ENTROPY_MODE_OFFSET, config_.entropy_mode},
+         {KMAC_CFG_SHADOWED_ENTROPY_FAST_PROCESS_BIT,
+          config_.entropy_fast_process},
+         {KMAC_CFG_SHADOWED_MSG_MASK_BIT, config_.msg_mask},
+         {KMAC_CFG_SHADOWED_ENTROPY_READY_BIT, config_.entropy_ready},
+         {KMAC_CFG_SHADOWED_ERR_PROCESSED_BIT, config_.err_processed},
+         {KMAC_CFG_SHADOWED_EN_UNSUPPORTED_MODESTRENGTH_BIT,
+          config_.en_unsupported_modestrength}});
+  }
+
+  void SetExpectedKey(const dif_kmac_key_t &key) {
+    std::map<dif_kmac_key_length_t, uint32_t> keySize = {
```

Comment:
key_size: variables are snake_case.

Separately, this map would be better off being a switch.

Reply:
Is there a guidance for snake_case and CamelCase?

Reply:
The style guide defers to the Google C++ style guide on this, for C++: https://docs.opentitan.org/doc/rm/c_cpp_coding_style/

Michael and I are talking about improving the guide to be more readable and cite-able, but most of my comments are grounded in something on it or what it effectively project-wide practice.